### PR TITLE
[solution]: zh/medium-append-argument.md (#191)

### PR DESCRIPTION
**Pull request by chenxiaoyao6228**
# Challenge [CHALLENGE_NAME]

## Checklist

- [x] Make sure that you are opening the PR with a single solution (one solution per PR)
- [x] The solution must have at least several references for the readers to dive in depth

### CODE REVIEW

**Comment by chenxiaoyao6228** [comment]
I'm really glad to contribute to this repo, but who should I ask to review my translation, any reply would be appreciated @ghaiklor

File: zh/medium-append-argument.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+让我们从推断函数参数及其返回类型开始, 条件类型将帮助我们完成这点。一旦类型被推断出来，我们可以返回我们自己输入的函数签名，举个栗子:
+
+```ts
+type AppendArgument<Fn, A> = Fn extends (args: infer P) => infer R
+  ? (args: P) => R
+  : never;
+```
+
+显然, 目前的答案还没达到我们想要的结果. 为什么呢? 因为我们检查了 `Fn` 是否可以赋值给具有单个参数 `args` 的函数。这并不合理，我们可以有多个参数或没有参数的函数。
+
```
Comment (likui628) [inline zh/medium-append-argument.md:35]: missing : 
> To fix that, we can use spread parameters:

File: zh/medium-append-argument.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+lang: en
+level: medium
+tags: arguments
+---
+
+## 挑战
+
+对于给定的函数类型`Fn`和任何类型`A`（在此上下文中,任何类型意味着我们不限制类型，并且我不考虑任何类型）创建一个将`Fn`作为第一个参数，`A`作为第二个参数的泛型函数，将生成函数类型`G`，它与`Fn`相同，但附加参数`A`作为最后一个参数。
+
+举个栗子:
```
Comment (likui628) [inline zh/medium-append-argument.md:13]: how about ‘例如’ ?

File: zh/medium-append-argument.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+现在，条件类型中的条件求值为真，因此进入带有类型参数`P`（函数参数）和类型参数`R`（返回类型）的`真`分支。虽然如此，目前方案还是有问题。类型参数 `P` 有一个带有函数参数的元组，但我们需要将它们视为单独的参数。通过应用可变元组类型，我们可以展开元组:
+
+```ts
+type AppendArgument<Fn, A> = Fn extends (...args: [...infer P]) => infer R
+  ? (args: P) => R
+  : never;
+```
+
+类型参数`P`有我们现在需要的。唯一剩下的事情就是从推断的类型中构造出新的函数签名:
```
Comment (likui628) [inline zh/medium-append-argument.md:50]: missing blank line

File: zh/medium-flatten.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+id: 459
+title: Flatten
+lang: en
+level: medium
+tags: array
+---
+
+## 挑战
+
+在这个挑战中，你需要编写一个接受数组并返回拍平数组类型的泛型函数, 举个栗子:
```
Comment (likui628) [inline zh/medium-flatten.md:11]: flatten : 展平
举个栗子：例如

File: zh/medium-flatten.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+```ts
+type flatten = Flatten<[1, 2, [3, 4], [[[5]]]> // [1, 2, 3, 4, 5]
+```
+
+## 方案
```
Comment (likui628) [inline zh/medium-flatten.md:17]: you'd better be consistent in translations.

File: zh/medium-flatten.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+该挑战的基本用例是一个空数组的情况。当我们输入一个空数组时，我们返回一个空数组， 反正它已经被拍平了。否则，返回`T` 本身:
+
+```typescript
+type Flatten<T> = T extends [] ? [] : T;
+```
+
+但是，如果 `T` 不是一个空数组，这意味着我们有一个包含元素的数组或元素本身。当它是一个包含元素的数组时，我们需要做什么？我们需要从它和尾巴中推断出其中一项。现在，我们可以只返回推断的项:
```
Comment (likui628) [inline zh/medium-flatten.md:25]: 尾巴中 -> 尾部

**Review by likui628 — COMMENTED** [review]
Thanks for your translation.
I suggest you PR one by one.

**Comment by chenxiaoyao6228** [comment]
I will close this PR, fix some translation errors and typos, then PR seperately

**Pull request closed without merging**